src/devices.rs

```diff
@@ -73,6 +73,10 @@ macro_rules! new_device {
                 })
             }
 
+            pub unsafe fn from_raw(raw: RawDevice) -> Self {
```

Comment:
I've removed it locally and all seems to work.

Reply:
Oh, doesn't _need_ to be there, but I put it in as a safeguard because it's essentially the same as a transmute: the user is saying "Yes, I'm sure that this is a XXX device, give me the correct traits."

I can go with a strongly-worded doc comment instead :)

Reply:
That would be ace, thanks. It's the same issue for all the device specific constructors. They should only be used if the user knows the type of the device for definite.